### PR TITLE
Provide stats for the recently introduced HTTP/2 rate limits

### PR DESCRIPTION
**Pull request by jvgutierrez**
Provide stats for the limits introduced by 9c09dbc11

### CODE REVIEW

**Comment by jvgutierrez** [comment]
> Overall looks great to me, but it seems like HTTP2_STAT_INSUFFICIENT_AVG_WINDOW_UPDATE_RATE is not incremented.

Fixed!

File: proxy/http2/HTTP2.h [hunk 1]
```diff
@@ -84,6 +84,12 @@ enum {
   HTTP2_STAT_SESSION_DIE_EOS,
   HTTP2_STAT_SESSION_DIE_ERROR,
   HTTP2_STAT_SESSION_DIE_HIGH_ERROR_RATE,
+  HTTP2_STAT_MAX_SETTINGS_PER_FRAME_EXCEEDED_RATE,
```
Comment (maskit) [inline proxy/http2/HTTP2.h]: The stats are not rates but counts, right? It may be just me, but I was confused a bit. Maybe we can omit "_RATE".

Comment (jvgutierrez) [reply proxy/http2/HTTP2.h]: hmmm right, they count the number of times that a specific rate limit has been exceeded, so we could drop the _RATE or reposition it.. like HTTP2_STAT_MAX_SETTINGS_PER_FRAME_RATE_EXCEEDED

Comment (maskit) [reply proxy/http2/HTTP2.h]: I'm also fine with the repositioning.

Comment (jvgutierrez) [reply proxy/http2/HTTP2.h]: I've dropped it after checking the naming, you were right :)

**Comment by zwoop** [comment]
Cherry-picked to v9.0.x branch.

**Pull request merged**